shared/actions/wallets.tsx

```diff
@@ -1194,12 +1255,99 @@ const searchTrustlineAssets = (state, {payload: {text}}) => {
         assets =>
           text === lastSearchText &&
           WalletsGen.createSetTrustlineSearchResults({
-            assets: assets ? assets.map(rpcAsset => rpcAssetToAssetDescription(rpcAsset)) : [],
+            assets: assets
+              ? (assets
+                  .map(rpcAsset => rpcAssetToAssetDescriptionOrNative(rpcAsset))
+                  .filter(asset => asset !== 'native') as Array<Types.AssetDescription>)
+              : [],
           })
       )
     : WalletsGen.createClearTrustlineSearchResults()
 }
 
+const paymentPathToRpcPaymentPath = (paymentPath: Types.PaymentPath): RPCStellarTypes.PaymentPath => ({
+  destinationAmount: paymentPath.destinationAmount,
+  destinationAsset: assetDescriptionOrNativeToRpcAsset(paymentPath.destinationAsset),
+  path: paymentPath.path.toArray().map(ad => assetDescriptionOrNativeToRpcAsset(ad)),
+  sourceAmount: paymentPath.sourceAmount,
+  sourceAmountMax: paymentPath.sourceAmountMax,
+  sourceAsset: assetDescriptionOrNativeToRpcAsset(paymentPath.sourceAsset),
+})
+
+const rpcPaymentPathToPaymentPath = (rpcPaymentPath: RPCStellarTypes.PaymentPath) =>
+  Constants.makePaymentPath({
+    destinationAmount: rpcPaymentPath.destinationAmount,
+    destinationAsset: rpcAssetToAssetDescriptionOrNative(rpcPaymentPath.destinationAsset),
+    path: I.List(
+      rpcPaymentPath.path
+        ? rpcPaymentPath.path.map(rpcAsset => rpcAssetToAssetDescriptionOrNative(rpcAsset))
+        : []
+    ),
+    sourceAmount: rpcPaymentPath.sourceAmount,
+    sourceAmountMax: rpcPaymentPath.sourceAmountMax,
+    sourceAsset: rpcAssetToAssetDescriptionOrNative(rpcPaymentPath.sourceAsset),
+  })
+
+const calculateBuildingAdvanced = state =>
+  RPCStellarTypes.localFindPaymentPathLocalRpcPromise(
+    {
+      amount: state.wallets.buildingAdvanced.recipientAmount,
+      destinationAsset: assetDescriptionOrNativeToRpcAsset(state.wallets.buildingAdvanced.recipientAsset),
+      from: state.wallets.buildingAdvanced.senderAccountID,
+      sourceAsset: assetDescriptionOrNativeToRpcAsset(state.wallets.buildingAdvanced.senderAsset),
+      to: state.wallets.buildingAdvanced.recipient,
+    },
+    Constants.calculateBuildingAdvancedWaitingKey
+  )
+    .then(res => {
+      console.log({res})
+      const {
+        destinationAccount,
+        destinationDisplay,
+        exchangeRate,
+        fullPath,
+        sourceDisplay,
+        sourceMaxDisplay,
+      } = res
+      return WalletsGen.createSetBuiltPaymentAdvanced({
+        builtPaymentAdvanced: Constants.makeBuiltPaymentAdvanced({
+          amountError: 'You only have 34.04 EUR/Stonghold.com available to spend.',
```

Comment:
We're not in a story file.. seems this shouldn't be hardcoded?

Reply:
Yup; I'll have a follow-up commit after some server/protocol change is in. Sorry for forgetting to leave a comment.